synapse/lib/view.py

```diff
@@ -603,6 +602,14 @@ async def setViewInfo(self, name, valu):
                 mesg = 'You may not set parent on a view which has more than one layer.'
                 raise s_exc.BadArg(mesg=mesg)
 
+        return await self._push('view:set', name, valu)
+
+    @s_nexus.Pusher.onPush('view:set')
+    async def _setViewInfo(self, name, valu):
+
+        if name == 'parent':
```

Comment:
We usually prefer as many safety checks as possible within the handler. Couldn't we just check if self.parent already equals the new parent and the existing one would be replay safe?